File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/ECGetResponse.cs
#region "Imports"

using System;
using System.Collections;
using PayPal.Payments.Common.Exceptions; 
using PayPal.Payments.Common.Utility ; 
#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Used for ExpressCheckout get operation.
	/// </summary>
	/// <remarks>
	/// <seealso cref="ExpressCheckoutResponse"/>
	/// <seealso cref="ECDoResponse"/>
	/// </remarks>
	public class ECGetResponse : ExpressCheckoutResponse
	{
		#region "Member variables"

		private String mEMail;
		private String mPayerId;
		private String mPayerStatus;
		private String mShipToFirstName;
		private String mShipToLastName;
		private String mShipToCountryCode;
		private String mShipToBusiness;
		private String mAddrStatus;
		private String mFirstName;
		private String mLastName;
		private String mStreet;
		private String mStreet2;
		private String mCity;
		private String mState;
		private String mPostalCode;
		private String mCountryCode;
		private String mPhoneNum;
		private String mBA_Flag;

		#endregion

		#region "Constructor"
		/// <summary>
		/// constructor
		/// </summary>
		internal ECGetResponse()
		{
		}
		#endregion

		#region "properties"
		/// <summary>
		/// Gets the EMail parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>EMAIL</code>
		/// </remarks>
		public String EMail
		{
			get {return mEMail;}
		}
		/// <summary>
		/// Gets the payerid parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>PAYERID</code>
		/// </remarks>
		public String PayerId
		{
			get {return mPayerId;}
		}

		/// <summary>
		/// Gets the payerstatus parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>PAYERSTATUS</code>
		/// </remarks>
		public String PayerStatus
		{
			get {return mPayerStatus;}
		}
		/// <summary>
		/// Gets the shiptofirstname parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOFIRSTNAME</code>
		/// </remarks>
		public String ShipToFirstName
		{
			get {return mShipToFirstName;}
		}
		/// <summary>
		/// Gets the shiptolastname parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOLASTNAME</code>
		/// </remarks>
		public String ShipToLastName
		{
			get {return mShipToLastName;}
		}
		/// <summary>
		/// Gets the ShipToCountry parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOCOUNTRY</code>
		/// </remarks>
		public String ShipToCountry
		{
			get {return mShipToCountryCode;}
		}
		/// <summary>
		/// Gets the ShipToBusiness parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOBUSINESS</code>
		/// </remarks>
		public String ShipToBusiness
		{
			get {return mShipToBusiness;}
		}
		/// <summary>
		/// Gets the AddressStatus parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>ADDRSTATUS</code>
		/// </remarks>
		public String AddressStatus
		{
			get {return mAddrStatus;}
		}
		/// <summary>
		/// Gets the FirstName parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>FIRSTNAME</code>
		/// </remarks>
		public String FirstName
		{
			get {return mFirstName;}
		}
		/// <summary>
		/// Gets the LastName parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>LASTNAME</code>
		/// </remarks>
		public String LastName
		{
			get {return mLastName;}
		}
		/// <summary>
		/// Gets the ShipToStreet parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOSTREET</code>
		/// </remarks>
		public String ShipToStreet
		{
			get {return mStreet;}
		}
		/// <summary>
		/// Gets the ShipToStreet2 parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOSTREET2</code>
		/// </remarks>
		public String ShipToStreet2
		{
			get {return mStreet2;}
		}
		/// <summary>
		/// Gets the ShipToCity parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOCITY</code>
		/// </remarks>
		public String ShipToCity
		{
			get {return mCity;}
		}
		/// <summary>
		/// Gets the ShipToState parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOSTATE</code>
		/// </remarks>
		public String ShipToState
		{
			get {return mState;}
		}
		/// <summary>
		/// Gets the ShipToZip parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOZIP</code>
		/// </remarks>
		public String ShipToZip
		{
			get {return mPostalCode;}
		}
		/// <summary>
		/// Gets the CountryCode parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>COUNTRYCODE</code>
		/// </remarks>
		public String CountryCode
		{
			get {return mCountryCode;}
		}
		/// <summary>
		/// Gets the PHONENUM parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>PHONENUM</code>
		/// </remarks>
		public String PhoneNum
		{
			get {return mPhoneNum;}
		}
		/// <summary>
		/// Gets the BA_FLAG parameter
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BA_FLAG</code>
		/// </remarks>
		public String BA_Flag
		{
			get {return mBA_Flag;}
		}

		#endregion

		#region "Core functions"

		/// <summary>
		/// Sets Response params
		/// </summary>
		///<param name="ResponseHashTable">Response Hash table by ref</param>
		internal override void SetParams(ref Hashtable ResponseHashTable)
		{
			try
			{
				mEMail = (String) ResponseHashTable[PayflowConstants.PARAM_EMAIL];
				mPayerId = (String) ResponseHashTable[PayflowConstants.PARAM_PAYERID];
				mPayerStatus = (String) ResponseHashTable[PayflowConstants.PARAM_PAYERSTATUS];
				mShipToFirstName = (String) ResponseHashTable[PayflowConstants.PARAM_SHIPTOFIRSTNAME];
				mShipToLastName = (String) ResponseHashTable[PayflowConstants.PARAM_SHIPTOLASTNAME];
				mShipToCountryCode = (String) ResponseHashTable[PayflowConstants.PARAM_SHIPTOCOUNTRY];
				mShipToBusiness = (String) ResponseHashTable[PayflowConstants.PARAM_SHIPTOBUSINESS];
				mAddrStatus = (String) ResponseHashTable[PayflowConstants.PARAM_ADDRSTATUS];
				mFirstName = (String) ResponseHashTable[PayflowConstants.PARAM_FIRSTNAME];
				mLastName = (String) ResponseHashTable[PayflowConstants.PARAM_LASTNAME];
				mStreet = (String) ResponseHashTable[PayflowConstants.PARAM_SHIPTOSTREET];
				mStreet2 = (String) ResponseHashTable[PayflowConstants.PARAM_SHIPTOSTREET2];
				mCity = (String) ResponseHashTable[PayflowConstants.PARAM_SHIPTOCITY];
				mState = (String) ResponseHashTable[PayflowConstants.PARAM_SHIPTOSTATE];
				mPostalCode = (String) ResponseHashTable[PayflowConstants.PARAM_SHIPTOZIP];
				mCountryCode = (String) ResponseHashTable[PayflowConstants.PARAM_COUNTRYCODE];
				mPhoneNum = (String) ResponseHashTable[PayflowConstants.PARAM_PHONENUM];
				mBA_Flag = (String) ResponseHashTable[PayflowConstants.PARAM_BA_FLAG];


				ResponseHashTable.Remove(PayflowConstants.PARAM_EMAIL);
				ResponseHashTable.Remove(PayflowConstants.PARAM_PAYERID);
				ResponseHashTable.Remove(PayflowConstants.PARAM_PAYERSTATUS);
				ResponseHashTable.Remove(PayflowConstants.PARAM_SHIPTOFIRSTNAME);
				ResponseHashTable.Remove(PayflowConstants.PARAM_SHIPTOLASTNAME);
				ResponseHashTable.Remove(PayflowConstants.PARAM_SHIPTOCOUNTRY);
				ResponseHashTable.Remove(PayflowConstants.PARAM_SHIPTOBUSINESS);
				ResponseHashTable.Remove(PayflowConstants.PARAM_ADDRSTATUS);
				ResponseHashTable.Remove(PayflowConstants.PARAM_FIRSTNAME);
				ResponseHashTable.Remove(PayflowConstants.PARAM_LASTNAME);
				ResponseHashTable.Remove(PayflowConstants.PARAM_SHIPTOSTREET);
				ResponseHashTable.Remove(PayflowConstants.PARAM_SHIPTOSTREET2);
				ResponseHashTable.Remove(PayflowConstants.PARAM_SHIPTOCITY);
				ResponseHashTable.Remove(PayflowConstants.PARAM_SHIPTOSTATE);
				ResponseHashTable.Remove(PayflowConstants.PARAM_SHIPTOZIP);
				ResponseHashTable.Remove(PayflowConstants.PARAM_COUNTRYCODE);
				ResponseHashTable.Remove(PayflowConstants.PARAM_PHONENUM);
				ResponseHashTable.Remove(PayflowConstants.PARAM_BA_FLAG);
			}
			catch (BaseException)
			{
				throw;
			}
			catch (Exception Ex)
			{
				DataObjectException DEx = new DataObjectException(Ex);
				throw DEx;
			}
			//catch
			//{
			//    throw new Exception();				
			//}

		}

		#endregion

	}
}

File: java/src/sdk/dataobjects/paypal/payflow/ECGetResponse.java
/*
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package paypal.payflow;

import paypal.payflow.ExpressCheckoutResponse;

import java.util.Hashtable;

/**
 * Used for ExpressCheckout update operation.
 * {@link ExpressCheckoutResponse}
 */

public class ECGetResponse extends ExpressCheckoutResponse {
    private String email;
    private String payerId;
    private String payerStatus;
    private String shipToFirstName;
    private String shipToLastName;
    private String shipToName;
    private String shipToCountryCode;
    private String shipToBusiness;
    //private String firstName;
    //private String lastName;
    private String shipToStreet;
    private String shipToStreet2;
    private String shipToCity;
    private String shipToState;
    private String shipToZip;
    private String countryCode;
    private String phoneNum;
    private String ba_Flag;
    private String street;
    private String street2;
    private String city;
    private String state;
    private String zip;
    private String addressStatus;

    /**
     * Gets the email parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: EMAIL
     */
    public String getEmail() {
        return email;
    }

    /**
     * Gets the payerid parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: PAYERID
     */
    public String getPayerId() {
        return payerId;
    }

    /**
     * Gets the payerstatus parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: PAYERSTATUS
     */
    public String getPayerStatus() {
        return payerStatus;
    }

    /**
     * Gets the shiptoname parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: SHIPTONAME
     */
    public String getShipToName() {
        return shipToName;
    }

    /**
     * Gets the shiptofirstname parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: SHIPTOFIRSTNAME
     */
    public String getShipToFirstName() {
         return shipToFirstName;
    }

    /**
     * Gets the shiptolastname parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: SHIPTOLASTNAME
     */
    public String getShipToLastName() {
         return shipToLastName;
    }

    /**
     * Gets the shiptocountry parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: SHIPTOCOUNTRY
     */
    public String getShipToCountry() {
        return shipToCountryCode;
    }

    /**
     * Gets the shiptobusiness parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: SHIPTOBUSINESS
     */
    public String getShipToBusiness() {
        return shipToBusiness;
    }

    /**
     * Gets the firstname parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: FIRSTNAME
     */
    //public String getFirstName() {
      //  return firstName;
    //}

    /**
     * Gets the lastname parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: LASTNAME
     */
    //public String getLastName() {
      //  return lastName;
   // }

    /**
     * Gets the shiptostreet parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: SHIPTOSTREET
     */
    public String getShipToStreet() {
        return shipToStreet;
    }

    /**
     * Gets the shiptostreet2 parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: SHIPTOSTREET2
     */
    public String getShipToStreet2() {
        return shipToStreet2;
    }

    /**
     * Gets the shiptocity parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: SHIPTOCITY
     */
    public String getShipToCity() {
        return shipToCity;
    }

    /**
     * Gets the shiptostate parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: SHIPTOSTATE
     */
    public String getShipToState() {
        return shipToState;
    }

    /**
     * Gets the countrycode parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: COUNTRYCODE
     */
    public String getCountryCode() {
        return countryCode;
    }

    /**
     * Gets the shiptozip parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: SHIPTOZIP
     */
    public String getShipToZip() {
        return shipToZip;
    }

    /**
     * Gets the ba_flag parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: BA_FLAG
     */
    public String getba_Flag() {
        return ba_Flag;
    }

    /**
     * Gets the phonenum parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: PHONENUM
     */
    public String getPhoneNum() {
        return phoneNum;
    }

    /**
     * Gets the street parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: STREET
     */
    public String getStreet() {
        return street;
    }

    /**
     * Gets the street2 parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: STREET2
     */
    public String getStreet2() {
        return street2;
    }

    /**
     * Gets the city parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: CITY
     */
    public String getCity() {
        return city;
    }

    /**
     * Gets the state parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: STATE
     */
    public String getState() {
        return state;
    }

    /**
     * Gets the zip parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: ZIP
     */
    public String getZip() {
        return zip;
    }

    /**
     * Gets the addressStatus parameter
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: ADDRESSSTATUS
     */
    public String getAddressStatus() {
        return addressStatus;
    }

    protected ECGetResponse() {
    }

    protected void setParams(Hashtable ResponseHashTable) {
        email = (String) ResponseHashTable.get(PayflowConstants.PARAM_EMAIL);
        payerId = (String) ResponseHashTable.get(PayflowConstants.PARAM_PAYERID);
        payerStatus = (String) ResponseHashTable.get(PayflowConstants.PARAM_PAYERSTATUS);
        shipToName = (String) ResponseHashTable.get(PayflowConstants.PARAM_SHIPTONAME);
        // 04/08/2015 shipToFirstName and shipToLastName are not returned in the response, mapping only.
        shipToFirstName = (String) ResponseHashTable.get(PayflowConstants.PARAM_FIRSTNAME);
        shipToLastName = (String) ResponseHashTable.get(PayflowConstants.PARAM_LASTNAME);
        shipToCountryCode = (String) ResponseHashTable.get(PayflowConstants.PARAM_SHIPTOCOUNTRY);
        shipToBusiness = (String) ResponseHashTable.get(PayflowConstants.PARAM_SHIPTOBUSINESS);
        //firstName = (String) ResponseHashTable.get(PayflowConstants.PARAM_FIRSTNAME);
        //lastName = (String) ResponseHashTable.get(PayflowConstants.PARAM_LASTNAME);
        street = (String) ResponseHashTable.get(PayflowConstants.PARAM_STREET);
        street2 = (String) ResponseHashTable.get(PayflowConstants.PARAM_STREET2);
        city = (String) ResponseHashTable.get(PayflowConstants.PARAM_CITY);
        state = (String) ResponseHashTable.get(PayflowConstants.PARAM_STATE);
        zip = (String) ResponseHashTable.get(PayflowConstants.PARAM_ZIP);
        shipToStreet = (String) ResponseHashTable.get(PayflowConstants.PARAM_SHIPTOSTREET);
        shipToStreet2 = (String) ResponseHashTable.get(PayflowConstants.PARAM_SHIPTOSTREET2);
        shipToCity = (String) ResponseHashTable.get(PayflowConstants.PARAM_SHIPTOCITY);
        shipToState = (String) ResponseHashTable.get(PayflowConstants.PARAM_SHIPTOSTATE);
        shipToZip = (String) ResponseHashTable.get(PayflowConstants.PARAM_SHIPTOZIP);
        countryCode = (String) ResponseHashTable.get(PayflowConstants.PARAM_COUNTRYCODE);
        phoneNum = (String) ResponseHashTable.get(PayflowConstants.PARAM_PHONENUM);
        ba_Flag = (String) ResponseHashTable.get(PayflowConstants.PARAM_BA_FLAG);
        addressStatus = (String) ResponseHashTable.get(PayflowConstants.PARAM_ADDRESSSTATUS);

        ResponseHashTable.remove(PayflowConstants.PARAM_EMAIL);
        ResponseHashTable.remove(PayflowConstants.PARAM_PAYERID);
        ResponseHashTable.remove(PayflowConstants.PARAM_PAYERSTATUS);
        ResponseHashTable.remove(PayflowConstants.PARAM_SHIPTONAME);
        //ResponseHashTable.remove(PayflowConstants.PARAM_SHIPTOFIRSTNAME);
        //ResponseHashTable.remove(PayflowConstants.PARAM_SHIPTOLASTNAME);
        ResponseHashTable.remove(PayflowConstants.PARAM_SHIPTOCOUNTRY);
        ResponseHashTable.remove(PayflowConstants.PARAM_SHIPTOBUSINESS);
        ResponseHashTable.remove(PayflowConstants.PARAM_FIRSTNAME);
        ResponseHashTable.remove(PayflowConstants.PARAM_LASTNAME);
        ResponseHashTable.remove(PayflowConstants.PARAM_SHIPTOSTREET);
        ResponseHashTable.remove(PayflowConstants.PARAM_SHIPTOSTREET2);
        ResponseHashTable.remove(PayflowConstants.PARAM_SHIPTOCITY);
        ResponseHashTable.remove(PayflowConstants.PARAM_SHIPTOSTATE);
        ResponseHashTable.remove(PayflowConstants.PARAM_SHIPTOZIP);
        ResponseHashTable.remove(PayflowConstants.PARAM_COUNTRYCODE);
        ResponseHashTable.remove(PayflowConstants.PARAM_PHONENUM);
        ResponseHashTable.remove(PayflowConstants.PARAM_BA_FLAG);
        ResponseHashTable.remove(PayflowConstants.PARAM_STREET);
        ResponseHashTable.remove(PayflowConstants.PARAM_STREET2);
        ResponseHashTable.remove(PayflowConstants.PARAM_CITY);
        ResponseHashTable.remove(PayflowConstants.PARAM_STATE);
        ResponseHashTable.remove(PayflowConstants.PARAM_ZIP);
        ResponseHashTable.remove(PayflowConstants.PARAM_ADDRESSSTATUS);

    }
}

